marshalable-derive/src/lib.rs

```diff
@@ -66,21 +66,24 @@ fn derive_tpm_marshal_inner(input: DeriveInput) -> Result<TokenStream> {
             ));
         }
     };
-
     let expanded = quote! {
-        #pure_impl
-        // The generated impl.
-        impl Marshalable for #name  {
-            fn try_unmarshal(buffer: &mut UnmarshalBuf) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<Self> {
-                #unmarsh_text
-            }
+        #[doc(hidden)]
+        const _: () = {
+            use tpm2_rs_marshalable::__private::*;
+            #pure_impl
+            // The generated impl.
+            impl Marshalable for #name  {
+                fn try_unmarshal(buffer: &mut UnmarshalBuf) -> TpmRcResult<Self> {
+                    #unmarsh_text
+                }
 
-            fn try_marshal(&self, buffer: &mut [u8]) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<usize> {
-                let mut written: usize = 0;
-                #marsh_text;
-                Ok(written)
+                fn try_marshal(&self, buffer: &mut [u8]) -> TpmRcResult<usize> {
```

Comment:
The reason I prefer fully qualified names over uses is because it makes it much much easier to debug when full path is not provided, but with `use`, it i a bit harder and it requires reviewer to look at the final macro output to see if there is a missing `use`.

Reply:
I find the long names hard to read; but I don't mind either way, this isn't something I mind just leaving as-is; it felt clean to me, but if you disagree I'll change it back.

Reply:
I slightly prefer fully qualified names.